SalesforceCommerceCloud/pwa-kit
# [App Extensibility ⚙️] Update `Create App` to generate a typescript-minimal project using an extension (@W-15839862@)

# Description



The PR updates the generator to allow users to create a typescript-minimal project using the available PWA Kit extensions.

The placeholder "extension-sample" has been made public temporarily to allow the generator to use it for project creation. This placeholder extension will eventually be replaced by the first extension we decide to publish to NPM.

- [x] TODO: The `extension-sample-dos` extension was added for testing purposes and will be removed before merging the PR.

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- Added option to generator to generate typescript-minimal project with extensions.

# How to Test-Drive This PR

1. Run the generator locally 
```
node packages/pwa-kit-create-app/scripts/create-mobify-app-dev.js
```
2. Answer the questions:

```
? Do you want to generate a project using Application Extensibility or using the template-retail-react-app? Generate a project using Application Extensibility
? Which Application Extensions do you want to install? 
// Use the space bar to select both extensions:
- extension-sample
- extension-sample-dos
? Do you want to extract the Application Extension code? Yes
? What is the name of your Project? test
```

3. Start the generated project and verify both server and client extension content loads as expected 
    - http://localhost:3000/foo-page
    - http://localhost:3000/sample 
    - http://localhost:3000/foo-page-dos
    - http://localhost:3000/sample-dos

Repeat the same steps to generate a new project but choose not to extract the  Application Extension extension code. Verify the extensions folders like "extension-sample" don't exist in the generated project.

---

Test the "presets":

```
node packages/pwa-kit-create-app/scripts/create-mobify-app-dev.js --preset "app-extension-sample-extract"
```

```
node packages/pwa-kit-create-app/scripts/create-mobify-app-dev.js --preset "app-extension-sample-no-extract"
```


# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
